dom.bs

```diff
@@ -6497,6 +6515,20 @@ an optional null or string <var>prefix</var> (default null), and an optional nul
  <a lt="append an attribute">append</a> this <a>attribute</a> to <var>element</var>, and then
  return.
 
+ <ol>
+  <li><p>Set <var>attribute</var> to a new <a>attribute</a> whose <a for=Attr>namespace</a> is
+  <var>namespace</var>, <a for=Attr>namespace prefix</a> is <var>prefix</var>,
+  <a for=Attr>local name</a> is <var>localName</var> and <a for=Node>node document</a> is
+  <var>element</var>'s <a for=Node>node document</a>.
+
+  <li><p><a>Validate and set attribute value</a> <var>value</var> for <var>attribute</var> with
+  <var>element</var>.
+
+  <li><p><a lt="append an attribute">Append</a> <var>attribute</var> to <var>element</var>.
+
+  <li><p>Return.
+ </ol>
```

Comment:
This seems wrong. This should be behind the "If attribute is null" check above, presumably?

Reply:
However, I'm also not sure why this cannot be part of "append an attribute" which would give us some deduplication. If you do the validation before the attribute is appended to the element, it seems fine? What am I missing?

Reply:
In particular because in these cases we end up creating a new attribute and so if something throws, it would simply be as if that attribute was not created, right?

And it seems that the current algorithms miss the `toggleAttribute()` endpoint.

Reply:
If we edit append an attribute we might end up with duplicated calls to validate.

For example in the screenshot below you'll see that we need to validate before step 5, but then step 6 which appends would also call validate?

<img width="802" alt="Screenshot 2024-04-10 at 16 12 14" src="https://github.com/whatwg/dom/assets/32498324/61a0d6c6-e7e3-4a20-acfa-b816ed1d4dd1">

Reply:
Having said that we can reorder these steps so that step 4 is inside of 5's if so it becomes 5.1 and 5.2 is to replace?

And then we can append attr and handle validation inside of append?

(I've also just spotted it refers to a newAttr which doesn't exist in this algorithm I'll make sure to fix that)

Reply:
Is `toggleAttribute()` actually an issue? Don't you still need to go through one of the other algorithms to actually set a value?

Reply:
Addressed in https://github.com/whatwg/dom/pull/1268